### PR TITLE
Legacytest

### PR DESCRIPTION
**Pull request by Dedlipid**
started writing tests for some features already written, included is pagination in the heroes module.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -21,5 +21,6 @@ module.exports = {
     '@typescript-eslint/explicit-function-return-type': 'off',
     '@typescript-eslint/explicit-module-boundary-types': 'off',
     '@typescript-eslint/no-explicit-any': 'off',
+    '@typescript-eslint/no-unused-vars': 'off',
```
Comment (hossein-zamanian) [inline .eslintrc.js:24]: why you do this ?

File: package.json [hunk 2]
```diff
@@ -43,7 +43,7 @@
     "@types/supertest": "^6.0.0",
     "@typescript-eslint/eslint-plugin": "^8.0.0",
     "@typescript-eslint/parser": "^8.0.0",
-    "eslint": "^9.0.0",
+    "eslint": "^8.57.1",
```
Comment (hossein-zamanian) [inline package.json:46]: why ?

File: src/leagues/heroes/heroes.service.ts [hunk 3]
```diff
@@ -31,15 +32,13 @@ export class LeaguesHeroesService {
       await queryRunner.manager.save(hero);
 
       await queryRunner.commitTransaction();
-    } catch (error) {
+    } catch {
       await queryRunner.rollbackTransaction();
```
Comment (hossein-zamanian) [inline src/leagues/heroes/heroes.service.ts:36]: why you ignore the error ?

File: src/leagues/heroes/heroes.service.ts [hunk 4]
```diff
@@ -59,7 +59,7 @@ export class LeaguesHeroesService {
           await queryRunner.manager.delete(League, heroOldLeague.id);
 
         await queryRunner.commitTransaction();
-      } catch (error) {
```
Comment (hossein-zamanian) [inline src/leagues/heroes/heroes.service.ts:62]: same as above issue

**Review by hossein-zamanian — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (Dedlipid) [reply .eslintrc.js:24]: otherwise it complains about stuff in the json module since they dont get used by anything now.

File: package.json (hunk 2 above)
Comment (Dedlipid) [reply package.json:46]: it was either downgrade the eslint or re write the eslint file, seemed simpler to downgrade.

**Comment by Dedlipid** [comment]
lint change explained and error throwing fixed.

**Pull request merged**